mbed_host_tests/host_tests_conn_proxy/conn_primitive_serial.py

```diff
@@ -84,6 +87,9 @@ def reset_dev_via_serial(self, delay=1):
 
     def read(self, count):
         """! Read data from serial port RX buffer """
+        # TIMEOUT: Since read is called in a loop, wait for self.timeout period before calling serial.read(). See
+        # comment on serial.Serial() call above about timeout.
+        sleep(self.timeout)
```

Comment:
Based on the relation between baud rate and bytes per sec for maximum baud rate of 115200 the max data rate would be ```11520``` Bps. For ```0.2``` sec receive window it will be ```2304``` bytes.  This tells we can only receive maximum ```2304``` bytes every ```0.2``` seconds. Hence if we have fear of drifting then we just have to increase our receive buffer to ```2304``` or more and pledge to process that before next ```0.2``` sec tick.